nikola/plugins/task/indexes.py

```diff
@@ -230,11 +233,16 @@ def index_path(self, name, lang, is_feed=False):
             index_file = os.path.splitext(self.site.config['INDEX_FILE'])[0] + extension
         else:
             index_file = self.site.config['INDEX_FILE']
+        if lang in self.number_of_pages:
+            number_of_pages = self.number_of_pages[lang]
+        else:
+            number_of_pages = self._compute_number_of_pages(self._get_filtered_posts(lang, self.site.config['SHOW_UNTRANSLATED_POSTS']), self.site.config['INDEX_DISPLAY_POST_COUNT'])
+            self.number_of_pages[lang] = number_of_pages
         return utils.adjust_name_for_index_path_list([_f for _f in [self.site.config['TRANSLATIONS'][lang],
                                                                     self.site.config['INDEX_PATH'],
                                                                     index_file] if _f],
                                                      name,
-                                                     utils.get_displayed_page_number(name, self.number_of_pages[lang], self.site),
+                                                     utils.get_displayed_page_number(name, self.number_of_pages, self.site),
```

Comment:
Ah, `self` should go. But then, the arguments aren't needed anyway since for the given arguments, `utils.adjust_name_for_index_path_list` simply returns the first argument.
(In particular, computing the page numbers isn't needed for sections.)